src/Network/Network/AzureFirewall/NewAzureFirewallCommand.cs

```diff
@@ -154,6 +154,24 @@ public class NewAzureFirewallCommand : AzureFirewallBaseCmdlet
         )]
         public string[] PrivateRange { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Flag to indicate DNS Proxy is enabled or disabled"
+        )]
+        public SwitchParameter EnableDnsProxy { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Flag to indicate if DNS Proxy is required for Network Rules"
+        )]
+        public SwitchParameter DisableDnsProxyForNetworkRule { get; set; }
```

Comment:
How about
```suggestion
        public SwitchParameter DnsProxyNotRequiredForNetworkRule { get; set; }
```
or even add an alias to shorten the name
```powershell
        [Alias("DnsProxyNotRequired")]
```

Reply:
Alias does not make it explicit. Changed to the proposed value.